lib/jsdom/browser/Window.js

```diff
@@ -183,23 +186,42 @@ function Window(options) {
   ///// METHODS
 
   let latestTimerId = 0;
+  let latestAnimationFrameCallbackId = 0;
 
   this.setTimeout = function (fn, ms) {
     const args = [];
     for (let i = 2; i < arguments.length; ++i) {
       args[i - 2] = arguments[i];
     }
-    return startTimer(window, setTimeout, clearTimeout, ++latestTimerId, fn, ms, args);
+    return startTimer(window, setTimeout, clearTimeout, ++latestTimerId, fn, ms, window.__timers, args);
   };
   this.setInterval = function (fn, ms) {
     const args = [];
     for (let i = 2; i < arguments.length; ++i) {
       args[i - 2] = arguments[i];
     }
-    return startTimer(window, setInterval, clearInterval, ++latestTimerId, fn, ms, args);
+    return startTimer(window, setInterval, clearInterval, ++latestTimerId, fn, ms, window.__timers, args);
+  };
+  this.requestAnimationFrame = fn => {
```

Comment:
Can you add a test to the to-upstream web platform tests that tests non-interference between setTimeout/setInterval and requestAnimationFrame? Basically:

- requestAnimationFrame's return value starts from zero, no matter how many setTimeouts/setIntervals happen before it.
- clearTimeout does not clear a requestAnimationFrame
- clearInterval does not clear a requestAnimationFrame
- cancelAnimationFrame does not clear a timeout
- cancelAnimationFrame does not clear an interval

Reply:
Done (I think 🙂)